dlp/snippets/inspect/inspect_bigquery_with_sampling.go

```diff
@@ -0,0 +1,185 @@
+	storageConfig := &dlppb.StorageConfig{
+		Type: &dlppb.StorageConfig_BigQueryOptions{
+			BigQueryOptions: bigQueryOptions,
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+	infoTypes := []*dlppb.InfoType{
+		{Name: "PERSON_NAME"},
+	}
+
+	// Specify how the content should be inspected.
+	inspectConfig := &dlppb.InspectConfig{
+		InfoTypes:    infoTypes,
+		IncludeQuote: true,
+	}
+
+	// Create a PubSub Client used to listen for when the inspect job finishes.
+	pubsubClient, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	defer pubsubClient.Close()
+
+	// Create a PubSub subscription we can use to listen for messages.
+	// Create the Topic if it doesn't exist.
+	t := pubsubClient.Topic(topicID)
+	if exists, err := t.Exists(ctx); err != nil {
+		fmt.Fprintf(w, "t.Exists: %v", err)
+		return err
+	} else if !exists {
+		if t, err = pubsubClient.CreateTopic(ctx, topicID); err != nil {
+			return err
+		}
+	}
+
+	// Create the Subscription if it doesn't exist.
+	s := pubsubClient.Subscription(subscriptionID)
+	if exists, err := s.Exists(ctx); err != nil {
+		return err
+	} else if !exists {
+		if s, err = pubsubClient.CreateSubscription(ctx, subscriptionID, pubsub.SubscriptionConfig{Topic: t}); err != nil {
+			return err
+		}
+	}
+
+	// topic is the PubSub topic string where messages should be sent.
+	topic := "projects/" + projectID + "/topics/" + topicID
+
+	action := &dlppb.Action{
+		Action: &dlppb.Action_PubSub{
+			PubSub: &dlppb.Action_PublishToPubSub{
+				Topic: topic,
+			},
+		},
+	}
+
+	// Configure the long running job we want the service to perform.
+	inspectJobConfig := &dlppb.InspectJobConfig{
+		StorageConfig: storageConfig,
+		InspectConfig: inspectConfig,
+		Actions: []*dlppb.Action{
+			action,
+		},
+	}
+
+	// Create the request for the job configured above.
+	req := &dlppb.CreateDlpJobRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/global", projectID),
+		Job: &dlppb.CreateDlpJobRequest_InspectJob{
+			InspectJob: inspectJobConfig,
+		},
+	}
+
+	// Use the client to send the request.
+	j, err := client.CreateDlpJob(ctx, req)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Job Created: %v", j.GetName())
+
+	// Wait for the inspect job to finish by waiting for a PubSub message.
+	// This only waits for 10 minutes. For long jobs, consider using a truly
+	// asynchronous execution model such as Cloud Functions.
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+	err = s.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+		// If this is the wrong job, do not process the result.
+		if msg.Attributes["DlpJobName"] != j.GetName() {
+			msg.Nack()
+			return
+		}
+		msg.Ack()
+
+		// Stop listening for more messages.
+		defer cancel()
+
+		resp, err := client.GetDlpJob(ctx, &dlppb.GetDlpJobRequest{
```

Comment:
Does it make sense to have this post-processing logic in the inner function? Seems like it would be easier to follow the error/return logic if it were not.

Reply:
Put the post-processing logic outside the inner function.